### PR TITLE
Added testing section and section on intercepting exceptions

### PR DESCRIPTION
**Pull request by GregOwen**
I also fixed the numbering to make it easier to modify (the explicit numbering is overridden by GitHub's markdown renderer anyways).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,7 +64,9 @@ Scala is an incredibly powerful language that is capable of many paradigms. We h
     - [Varargs](#java-varargs)
     - [Implicits](#java-implicits)
     - [Companion Objects, Static Methods and Fields](#java-companion-object)
-  6. [Miscellaneous](#misc)
+  1. [Tests] (#tests)
```
Comment (rxin) [inline README.md]: Tests -> Testing?

File: README.md [hunk 2]
```diff
@@ -1084,6 +1086,25 @@ There are a few things to watch out for when it comes to companion objects and s
   case object MyClass extends MyClass
   ```
 
+## <a name='tests'>Tests</a>
```
Comment (rxin) [inline README.md]: Testing

**Comment by rxin** [comment]
LGTM other than that.

**Comment by GregOwen** [comment]
@rxin Fixed

**Comment by rxin** [comment]
Actually can you also update history section?

**Comment by GregOwen** [comment]
D'oh. Added it.

**Comment by rxin** [comment]
Thanks.

**Comment by rxin** [comment]
cc @HyukjinKwon / @Hawstein want to translate? :)

**Comment by HyukjinKwon** [comment]
Sure, thanks for cc'ing me.

**Comment by Hawstein** [comment]
Sure

**Pull request merged**